frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -432,16 +442,40 @@ export default class NativeQueryEditor extends Component {
         </div>
         <ResizableBox
           ref="resizeBox"
-          className={"border-top " + editorClasses}
+          className={cx("border-top flex ", { hide: !isNativeEditorOpen })}
           height={this.state.initialHeight}
           minConstraints={[Infinity, getEditorLineHeight(MIN_HEIGHT_LINES)]}
           axis="y"
+          handle={<div className="NativeQueryEditorDragHandle" />}
           onResizeStop={(e, data) => {
             this.props.handleResize();
             this._editor.resize();
           }}
+          resizeHandles={["s"]}
         >
-          <div id="id_sql" ref="editor" />
+          <div className="flex-full" id="id_sql" ref="editor" />
+          <div className="flex flex-column align-center border-left">
+            {[DataReferenceButton, NativeVariablesButton].map(Button => (
+              <Button {...this.props} size={ICON_SIZE} className="mt3" />
+            ))}
+            <RunButtonWithTooltip
+              // The button disappears when it's not runnable.
```

Comment:
I'm starting to think we'll probably want to refactor or split apart this component at some point soon. It feels like we could almost just use a button here in the native editor context as opposed to the sometimes show sometimes don't statefulness that the old editor had. The props list is getting a bit out of hand.

Reply:
Yeah good point. It's a simple component, but its style varies a lot between the three places we use it.

This prop was particularly awkward because I lied about the value to get the right styling. I just dropped `isRunnable` in favor of using `hidden` or `disabled` as needed.